Data/models/Bookmark.swift

```diff
@@ -425,6 +426,55 @@ extension Bookmark {
         record?.delete()
     }
     
+    /// Gets all nested bookmarks recursively.
+    public static func getRecursiveChildren(forFolderUUID syncUUID: [Int]?,
+                                            context: NSManagedObjectContext = DataController.viewContext) -> [Bookmark]? {
+        guard let searchableUUID = SyncHelpers.syncDisplay(fromUUID: syncUUID) else {
+            return nil
+        }
+        
+        let syncParentDisplayUUIDKeyPath = #keyPath(Bookmark.syncParentDisplayUUID)
+        
+        let predicate = NSPredicate(format: "\(syncParentDisplayUUIDKeyPath) == %@", searchableUUID)
+        
+        var allBookmarks = [Bookmark]()
+        
+        let result = all(where: predicate, context: context)
+        
+        result?.forEach {
+            allBookmarks.append($0)
+            
+            if $0.isFolder {
+                if let nestedBookmarks = getRecursiveChildren(forFolderUUID: $0.syncUUID) {
+                    allBookmarks.append(contentsOf: nestedBookmarks)
+                }
+            }
+        }
+        
+        return allBookmarks
+    }
+    
+    public func removeFolderAndSendSyncRecords(uuid: [Int]?) {
```

Comment:
Is this needed? Can't the normal `remove` function take care of this? Rather than identifying which function to call?

Reply:
Normal `remove` drops folder and children all at once, while for sync, each record to delete must be passed separately.

Reply:
So after you drop folder, we have no children bookmark references to send them to sync's remove